sds-go/go/dd_sds.h

```diff
@@ -3,11 +3,19 @@
 #include <stdlib.h>
 #include <stdbool.h>
 
-long create_scanner(const char* rules_as_json, const char** error, bool should_keywords_match_event_paths);
+long create_regex_rule(const char* json_config);
+
+long create_scanner(long rule_list, const char** error, bool should_keywords_match_event_paths);
 void delete_scanner(long scanner_id);
 
 // event is a non-null terminated TODO
 const char* scan(long scanner_id, const void* event, long event_size, long *retsize, long *retcap, const char** error);
 
 void free_vec(const char* string, long len, long cap);
 void free_string(const char* string);
+
+void free_any_rule(long rule_ptr);
+
+long create_rule_list();
+void append_rule_to_list(long rule_ptr, long list_ptr);
+void free_rule_list(long list_ptr);
```

Comment:
If I read it correctly, all this logic is hidden from the library interface, and this is really good to keep our interface simple 👍

Reply:
Correct, when you create a `Scanner`, all of this is done automatically. You just pass a list of "rule configs", and the rules will be allocated / added to the list automatically for you